### ISSUE TITLE
Links not being added to wallabag?

### ISSUE THREAD

**Issue by mawmawmawm** [Findus23/rss2wallabag#42]
Hi there,
first of all thanks for writing this.
I'm struggling with the tool not adding links to wallabag although the debug log says links were added.

Here's what I did:
* `git clone https://github.com/Findus23/rss2wallabag.git`
* Installed requirements via `pip install -r requirements.txt`
* Renamed `config.example.yaml` and `sites.example.yaml` and put in credentials and feed.
* Ran tool via `python main.py`.

The debug output is as follows:

```
# python main.py
2019-02-10 23:16:05,016 - DEBUG - Using selector: EpollSelector
2019-02-10 23:16:06,204 - INFO - Downloading feed: ifun
2019-02-10 23:16:07,279 - INFO - Parsing feed: ifun
2019-02-10 23:16:07,434 - DEBUG - finished parsing: ifun
2019-02-10 23:16:07,434 - INFO - article found: Hardware-Angebote: Philips Hue, Xiaomi Roborock und Mac mini
2019-02-10 23:16:07,435 - INFO - add to wallabag: Hardware-Angebote: Philips Hue, Xiaomi Roborock und Mac mini
2019-02-10 23:16:08,053 - INFO - article found: Apple Karten: Hier werden demnächst Foto-Aufnahmen gemacht
2019-02-10 23:16:08,054 - INFO - add to wallabag: Apple Karten: Hier werden demnächst Foto-Aufnahmen gemacht
2019-02-10 23:16:08,634 - INFO - article found: Das Alexa-Archiv: Alte Sprachaufnahmen anhören und löschen
2019-02-10 23:16:08,636 - INFO - add to wallabag: Das Alexa-Archiv: Alte Sprachaufnahmen anhören und löschen
2019-02-10 23:16:09,199 - INFO - article found: Netzwerk im Blick: Aktueller Datendurchsatz in der Mac-Menüleiste
2019-02-10 23:16:09,200 - INFO - add to wallabag: Netzwerk im Blick: Aktueller Datendurchsatz in der Mac-Menüleiste
2019-02-10 23:16:09,763 - INFO - article found: Werbeblocker: Spotify verbietet den Einsatz ausdrücklich
2019-02-10 23:16:09,763 - INFO - add to wallabag: Werbeblocker: Spotify verbietet den Einsatz ausdrücklich
2019-02-10 23:16:10,325 - INFO - article found: macOS Dark Mode: Hell/Dunkel-Umschalter selbst gebaut
2019-02-10 23:16:10,326 - INFO - add to wallabag: macOS Dark Mode: Hell/Dunkel-Umschalter selbst gebaut
2019-02-10 23:16:10,889 - INFO - article found: Netflix: „Smart Downloads“ erreichen iPad und iPhone
2019-02-10 23:16:10,890 - INFO - add to wallabag: Netflix: „Smart Downloads“ erreichen iPad und iPhone
2019-02-10 23:16:11,447 - INFO - article found: LibreOffice 6.2: Neue Version der freien Word- und Excel-Nachbauten
2019-02-10 23:16:11,448 - INFO - add to wallabag: LibreOffice 6.2: Neue Version der freien Word- und Excel-Nachbauten
2019-02-10 23:16:12,015 - INFO - article found: FaceTime-Fix: Ergänzendes Update auch für macOS 10.14.3
2019-02-10 23:16:12,016 - INFO - add to wallabag: FaceTime-Fix: Ergänzendes Update auch für macOS 10.14.3
2019-02-10 23:16:12,574 - INFO - article found: iTunes 12.9: Link reaktiviert alte App Store-Suche
2019-02-10 23:16:12,575 - INFO - add to wallabag: iTunes 12.9: Link reaktiviert alte App Store-Suche
2019-02-10 23:16:13,134 - INFO - article found: Neuer Passwort-Manager für den Mac: Lokal und offline wie Pastor
2019-02-10 23:16:13,135 - INFO - add to wallabag: Neuer Passwort-Manager für den Mac: Lokal und offline wie Pastor
2019-02-10 23:16:13,703 - INFO - article found: Skype bringt „Background Blur“: Weichzeichnen statt aufräumen
2019-02-10 23:16:13,704 - INFO - add to wallabag: Skype bringt „Background Blur“: Weichzeichnen statt aufräumen
2019-02-10 23:16:14,261 - INFO - article found: Löschung aufgeschoben: Flickr verlängert Gnadenfrist
2019-02-10 23:16:14,262 - INFO - add to wallabag: Löschung aufgeschoben: Flickr verlängert Gnadenfrist
2019-02-10 23:16:14,830 - INFO - article found: Instaradio: Online-Radio-App startet neu
2019-02-10 23:16:14,831 - INFO - add to wallabag: Instaradio: Online-Radio-App startet neu
2019-02-10 23:16:15,392 - INFO - article found: WISO steuer:Mac zum Aktionspreis bei eBay Wow
2019-02-10 23:16:15,393 - INFO - add to wallabag: WISO steuer:Mac zum Aktionspreis bei eBay Wow
2019-02-10 23:16:15,959 - INFO - article found: Sky Q künftig mit Fußball-Highlights auf Abruf
2019-02-10 23:16:15,961 - INFO - add to wallabag: Sky Q künftig mit Fußball-Highlights auf Abruf
2019-02-10 23:16:16,520 - INFO - article found: Bundeskartellamt untersagt Facebook die Zusammenführung von Nutzerdaten
2019-02-10 23:16:16,521 - INFO - add to wallabag: Bundeskartellamt untersagt Facebook die Zusammenführung von Nutzerdaten
2019-02-10 23:16:17,084 - INFO - article found: Surface Headphones: Microsoft-Kopfhörer starten in Deutschland
2019-02-10 23:16:17,085 - INFO - add to wallabag: Surface Headphones: Microsoft-Kopfhörer starten in Deutschland
2019-02-10 23:16:17,648 - DEBUG - already added: DAZN bietet MultiView mit vier parallelen Streams auf Apple TV
#
```

Here's my config.yaml :
```
wallabag:
  host: https://[redacted]
  client_id: 4_[redacted]
  client_secret: [redacted]
  username: [redacted]
  password: [redacted]
# github_username: mawmawmawm
debug: true
```

And my sites.yaml:

```
ifun:
  latest_article: DAZN bietet MultiView mit vier parallelen Streams auf Apple TV
  tags:
  - ifun
  url: http://www.ifun.de/feed/

```

Everything seems to work, but I don't see any of the articles being added to wallabag.
I double checked my credentials, `client_id` and `client_secret` and the host.
There's a special character in my password though, could that be it? I relalized that if I change the password to something invalid the debug output is exactly the same.

Any ideas?
Thanks.

**Comment by Findus23** [comment]
Without taking a closer look, I guess you are right that maybe the password isn't read correctly.

Maybe try wrapping it in " 

You could also add a `print(config)` here to make sure the file is read correctly:
https://github.com/Findus23/rss2wallabag/blob/7d22387f6a242ef3c3af0d0b2ef4c9d0b62045a5/main.py#L26-L28

**Comment by mawmawmawm** [comment]
Thanks - I did that, the special chars are being recognized correctly with out the wrapping things in quotes. 

The actual mistake was something different however: The debug mode. I reviewed the code and [found this](https://github.com/Findus23/rss2wallabag/blob/7d22387f6a242ef3c3af0d0b2ef4c9d0b62045a5/main.py#L115) in there - the actual _add to wallabag_ is only being done if the script does not run in debug mode. This should be made clear (e.g. by an additional print `warning: running in debug mode - not adding links to wallabag` or so) somewhere since it's not obvious. :)
Thanks again.
